internals/secrethub/credential_list.go

```diff
@@ -0,0 +1,76 @@
+package secrethub
+
+import (
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
+)
+
+// CredentialListCommand creates a backup code to restore a credential from a code.
+type CredentialListCommand struct {
+	io            ui.IO
+	newClient     newClientFunc
+	useTimestamps bool
+}
+
+// NewAccountInitCommand creates a new CredentialListCommand.
+func NewCredentialListCommand(io ui.IO, newClient newClientFunc) *CredentialListCommand {
+	return &CredentialListCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *CredentialListCommand) Register(r command.Registerer) {
+	clause := r.Command("ls", "List all your credentials.")
+	clause.Alias("list")
+
+	registerTimestampFlag(clause).BoolVar(&cmd.useTimestamps)
+
+	command.BindAction(clause, cmd.Run)
+}
+
+// Run lists all the currently authenticated account's credentials.
+func (cmd *CredentialListCommand) Run() error {
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	credentials, err := client.Credentials().ListMine()
+	if err != nil {
+		return err
+	}
+
+	timeFormatter := NewTimeFormatter(cmd.useTimestamps)
+
+	w := tabwriter.NewWriter(cmd.io.Stdout(), 0, 2, 2, ' ', 0)
+	fmt.Fprintln(w,
+		"NAME\t"+
```

Comment:
Currently, this field is always set to `default credential` (which is the server default when no name is supplied). There's no way (yet) to supply this as a user.

Do we want to release listing the name before there's a way to set it?

See this related thread https://github.com/secrethub/secrethub-go/pull/150 on whether to have a name, description or device field.

Reply:
[`secrethub init` gives the option to supply a name.](https://github.com/secrethub/secrethub-cli/pull/219)